prototype.tf

```diff
@@ -191,6 +191,64 @@ resource "google_cloudfunctions_function" "data_ingestion_to_gcs" {
  * [END] GCF Setup
  */
 
+ /*
+  * [BEGIN] Cloud Run Setup
+  */
+
+# Service account used to invoke the cloud run service through the push subscription.
+resource "google_service_account" "ingestion_identity" {
+  account_id = var.ingestion_identity_id
+}
+
+# Push subscription for upload_to_gcs topic that invokes the run service.
+resource "google_pubsub_subscription" "ingestion_subscription" {
+  name = var.ingestion_subscription_name
+  topic = google_pubsub_topic.upload_to_gcs.name
+  
+  ack_deadline_seconds = 20
+  
+  push_config {
+    push_endpoint = google_cloud_run_service.ingestion_service.status.0.url
```

Comment:
Is 0 the task number? Maybe include a comment with a link to documentation for finding a cloud run endpoint (I couldn't find it in the terraform docs)

Reply:
Excellent question! In proto format, status is a single message not repeated. I didn't see anything in the terraform docs either. At first I thought this was a bug in terraform, but after some searching, I think it may be a quirk of the terraform configuration language. There doesn't seem to be an equivalent of a single nested message. They are either lists or maps or sets. So the 0 is just for the first element in the list of one.